src/main/java/com/example/chapter1/basicprogramming/ReverseArray.java

```diff
@@ -0,0 +1,42 @@
+package com.example.chapter1.basicprogramming;
+
+//TODO: I'm not in favor of static methods. If someone want's make it better please contribute.
+//TODO: I'm not able to make methods generic which accepts primitives. If someone make it better please contribute.
+//TODO: Logic seems to duplicated. Refactor it!!
+public class ReverseArray {
+
+
+    public static int[] reverse(int[] numericArray) {
+        int arrayLength = numericArray.length;
+        if (arrayLength == 0) throw new IllegalArgumentException("Empty array.");
+        for (int index = 0; index < arrayLength / 2; index++) {
+            int temp = numericArray[index];
+            numericArray[index] = numericArray[arrayLength - 1 - index];
+            numericArray[arrayLength - index - 1] = temp;
+        }
+        return numericArray;
+    }
+
+    public static double[] reverse(double[] numericArray) {
```

Comment:
For java we need to write implementations for each data type separately that is fine for now but how much we are going to write and the code seems ultimately duplicated.
Can we take advantage of java autoboxing somehow. Like instead of passing primitives we can have wrappers may be of type Number which can hold anything like integers and floats.